### PR TITLE
Refactor actions

### PR DESCRIPTION
**Pull request by Haud**
This is a refactor of the YoshiActionResult as discussed in PR 24:

* YoshiActionResult struct removed
* YoshiResult renamed into YoshiActionResult
* YoshiActionResult.PresentViewController now takes a UIViewController as an associated value
* YoshiActionResult convenience method to retrieve a UIViewController

### CODE REVIEW

File: Yoshi/Yoshi/YoshiActionResult.swift [hunk 1]
```diff
@@ -1,42 +1,53 @@
...
+            return block
+        default:
+            return { }
+        }
     }
 
-    public init(action: () -> Void) {
-        result = .AsyncAfterDismissing(action)
-        viewController = nil
+    func viewController() -> UIViewController? {
```
Comment (htinlinn) [inline Yoshi/Yoshi/YoshiActionResult.swift]: Do we really need these two functions and the `Equatable` implementation here? Isn't the Swift built-in pattern matching enough for our purpose? E.g.

```
        switch result {
        case .PresentViewController(let viewController):
            // do things to view controller
        case .AsyncAfterDismissing(let asyncBlock):
            // call or pass block
        default:
            return
        }
```

Comment (Haud) [inline Yoshi/Yoshi/YoshiActionResult.swift]: Good call. done.

**Comment by htinlinn** [comment]
:+1:

**Comment by MichaelCampbell** [comment]
:+1:

**Pull request merged**